File: Javascript/CodingProblems/MaxProfit.js

const findMaxProfit = (prices) => {
    let maxProfit = 0;
    let buy = prices[0];
    for(let i = 1; i < prices.length; i++){
        const sell = prices[i];
        if(sell > buy){
            const profit = sell - buy;
            if(profit > maxProfit){
                maxProfit = profit;
            }
        }else{
            buy = prices[i];
        }
    }
    return maxProfit;
};


const test1 = findMaxProfit([7, 1, 5, 3, 6, 4]);
console.log('test1', test1);


const test2 = findMaxProfit([7, 6, 4, 3, 1]);
console.log('test2', test2);

File: CSharp/CodingProblems/MaxProfit.cs

namespace DSA.CodingProblems
{
    // Leetcode #121
    public class MaxProfit
    {
        public int FindMaxProfit(int[] prices)
        {
            int maxProfit = 0;
            int buy = prices[0];
            for(int i = 1; i< prices.Length; i++)
            {
                var sell = prices[i];
                if (sell > buy)
                {
                    var profit = sell - buy;
                    if(profit > maxProfit)
                        maxProfit = profit;
                }
                else 
                { 
                    buy = prices[i];
                }
                
            }
            return maxProfit;
        }
    }
}
